processing/src/main/java/io/druid/segment/incremental/IncrementalIndexAdapter.java

```diff
@@ -301,4 +305,85 @@ public ColumnCapabilities getCapabilities(String column)
   {
     return index.getCapabilities(column);
   }
+
+  @Override
+  public BitmapIndexSeeker getBitmapIndexSeeker(String dimension)
+  {
+    final Map<String, MutableBitmap> dimInverted = invertedIndexes.get(dimension);
+    if (dimInverted == null) {
+      return new EmptyBitmapIndexSeeker();
+    }
+
+    return new BitmapIndexSeeker()
+    {
+      private String lastVal = null;
+
+      @Override
+      public IndexedInts seek(String value)
+      {
+        if (value != null && GenericIndexed.STRING_STRATEGY.compare(value, lastVal) <= 0)  {
+          throw new ISE("Value[%s] is less than the last value[%s] I have, cannot be.",
+              value, lastVal);
+        }
+        lastVal = value;
+        final MutableBitmap bitmapIndex = dimInverted.get(value);
+        if (bitmapIndex == null) {
+          return new EmptyIndexedInts();
+        }
+        return new IndexedInts()
+        {
+          @Override
+          public int size()
+          {
+            return bitmapIndex.size();
+          }
+
+          @Override
+          public int get(int index)
+          {
+            throw new UnsupportedOperationException("This is really slow, so it's just not supported.");
```

Comment:
@drcrallen, done, i do fill out a CLA.
@xvrl, i do refactor the code and reuse it.